src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/RetryLogic/SqlConnectionCRLTest.cs

```diff
@@ -0,0 +1,192 @@
+        public void ConnectionCancelRetryOpenInvalidCatalog(string cnnString, SqlRetryLogicBaseProvider provider)
+        {
+            int cancelAfterRetries = provider.RetryLogic.NumberOfTries - 1;
+            int currentRetries = 0;
+            provider.Retrying += (s, e) => currentRetries = e.RetryCount;
+            var cnn = CreateConnectionWithInvalidCatalog(cnnString, provider, cancelAfterRetries);
+            var ex = Assert.Throws<AggregateException>(() => cnn.Open());
+            Assert.Equal(cancelAfterRetries, currentRetries);
+            Assert.Equal(cancelAfterRetries, ex.InnerExceptions.Count);
+            Assert.Contains(string.Format(_cancelErrMsgPattern, currentRetries), ex.Message);
+        }
+
+        [ActiveIssue(14590, TestPlatforms.Windows)]
+        // avoid creating a new database in Azure
+        [ConditionalTheory(typeof(DataTestUtility), nameof(DataTestUtility.IsNotAzureServer))]
+        [MemberData(nameof(RetryLogicTestHelper.GetConnectionAndRetryStrategyLongRunner), parameters: new object[] { 10 }, MemberType = typeof(RetryLogicTestHelper))]
+        public void CreateDatabaseWhileTryingToConnect(string cnnString, SqlRetryLogicBaseProvider provider)
+        {
+            int currentRetries = 0;
+            string database = DataTestUtility.GetUniqueNameForSqlServer($"RetryLogic_{provider.RetryLogic.RetryIntervalEnumerator.GetType().Name}", false);
+            var builder = new SqlConnectionStringBuilder(cnnString)
+            {
+                InitialCatalog = database,
+                ConnectTimeout = 1
+            };
+
+            using (var cnn1 = new SqlConnection(new SqlConnectionStringBuilder(cnnString) { ConnectTimeout = 60, Pooling = false }.ConnectionString))
+            {
+                cnn1.Open();
+                Task createDBTask = null;
+                try
+                {
+                    provider.Retrying += (s, e) =>
+                    {
+                        currentRetries = e.RetryCount;
+                        using (var cmd = cnn1.CreateCommand())
+                        {
+                            // Try to create database just after first faliure.
+                            if (createDBTask == null || createDBTask.Status == TaskStatus.Faulted)
+                            {
+                                cmd.CommandText = $"IF (NOT EXISTS(SELECT 1 FROM sys.databases WHERE name = '{database}')) CREATE DATABASE [{database}];";
+                                createDBTask = cmd.ExecuteNonQueryAsync();
+                            }
+                        }
+                    };
+
+                    using (var cnn2 = new SqlConnection(builder.ConnectionString))
+                    {
+                        cnn2.RetryLogicProvider = provider;
+                        cnn2.Open();
+                    }
+                }
+                catch (Exception e)
+                {
+                    Assert.Null(e);
```

Comment:
I think you can avoid catching exceptions like this as otherwise it will throw: Assertion Failed errors if this happens, instead of the the actual error message being thrown which is more important to know.